autometa/taxonomy/ncbi.py

```diff
@@ -364,60 +347,68 @@ def parse_merged(self):
+        taxid_A_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_A)}
+        taxid_B_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_B)}
+        common_ancestor = (taxid_B_lineage_taxids.intersection(
+            taxid_A_lineage_taxids))
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given taxid is a positive interger or not 
 
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
 
-def main():
-    import argparse
-    import logging as logger
-    logger.basicConfig(
-        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
-        datefmt='%m/%d/%Y %I:%M:%S %p',
-        level=logger.DEBUG)
-    parser = argparse.ArgumentParser(
-        description='Autometa NCBI utilities class')
-    parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
-    parser.add_argument(
-        '--query-taxid',
-        help='query taxid to test NCBI class functionality',
-        default=1222,
-        type=int,
-    )
-    parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
+        Returns
+        -------
+        boolean
+            True if the 'taxid' is a positive interger  else False
+        """
+
+        # This check if an interger has been added as str, eg. "562"
+        if isinstance(taxid, str):
+            if taxid.isnumeric() or taxid.replace('.', '', 1).isnumeric():
+                # checks if it is something like 12.0 vs. 12.9.
+                if float(taxid).is_integer():
+                    # See https://stackoverflow.com/a/47764450/12671809
+                    taxid = int(float(taxid))
+                else:
+                    return False
+            else:
+                return False
+        # `is_integer` returns error if only '12' is used, float(taxid) is needed
+        if not float(taxid).is_integer() or taxid < 0:
+            return False
+        return True
```

Comment:
Previously the taxid as a `str` was being converted to an `int`. Jason commented on this as "casting" the provided `taxid` as an `int`. This functionality has been removed although we are similarly performing all of the checks. This either needs to be reworked. Or the other methods need to be updated so they can handle input taxids as `str`. My vote would be implementing this behavior in `self.is_valid_taxid`

Reply:
I think `taxid = int(float(taxid))` is converting `str` to `int`. Is this what you mean by "casting"?

Reply:
The _only_ returned values for this method are `True` and `False`. You may be converting `taxid` here, but this function does nothing with the converted `taxid`. What about the taxids in the methods that are being used?

See:

- https://en.wikipedia.org/wiki/Type_conversion
- [w3 tutorials python casting](https://www.w3schools.com/python/python_casting.asp)
- [type conversion python Geeks4Geeks](https://www.geeksforgeeks.org/type-conversion-python/)
- [Casting and type conversions (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions)